File: Source/Kernel/Contracts/EventSequences/HistogramBucket.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Chronicle.Contracts.EventSequences;

/// <summary>
/// Represents the number of events that occurred within one time bucket of an event sequence histogram.
/// </summary>
[ProtoContract]
public class HistogramBucket
{
    /// <summary>
    /// Gets or sets the inclusive start of the time bucket, truncated to the requested resolution.
    /// </summary>
    [ProtoMember(1)]
    public DateTimeOffset Occurred { get; set; }

    /// <summary>
    /// Gets or sets the number of events that occurred within the bucket.
    /// </summary>
    [ProtoMember(2)]
    public long Count { get; set; }
}

File: Source/Workbench/Features/EventStore/Namespaces/Sequences/QueryEditor/HistogramBucket.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

/**
 * One pre-counted bar of the time range histogram, on an epoch-millisecond scale.
 */
export interface HistogramBucket {
    /** Inclusive start of the bucket. */
    start: number;
    /** Exclusive end of the bucket. */
    end: number;
    /** Number of events that occurred within the bucket. */
    count: number;
}
